spec/models/user_spec.rb

```diff
@@ -25,5 +27,42 @@
 require 'rails_helper'
 
 RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  subject do
+    FactoryBot.build(:user, role: role)
+  end
+
+  let(:role) { FactoryBot.create(:role) }
+
+  describe 'Associations' do
+    it { is_expected.to belong_to(:role) }
+  end
+
+  it 'is valid with valid attributes' do
+    expect(subject).to be_valid
+  end
+
+  it 'is not valid without a email' do
+    subject.email = nil
+    expect(subject).not_to be_valid
```

Comment:
Esto se podría hace usan `shoulda-matchers` con:
```suggestion
    it { should validate_presence_of(:email) }
```

Reply:
Acá te dejo un ejemplo de una validacion de modelo usando `factory bot` y `shoulda matchers`:
[https://github.com/thoughtbot/shoulda-matchers#on-the-subject-of-subject](https://github.com/thoughtbot/shoulda-matchers#on-the-subject-of-subject)